lib/models/environment-variables.coffee

```diff
@@ -143,3 +143,126 @@ exports.remove = (id, callback) ->
 ####
 exports.isSystemVariable = (variable) ->
 	return /^RESIN_/.test(variable.name)
+
+exports.device = {}
+
+###*
+# device.getAll callback
+# @callback module:resin.models.environment-variables.device~getAllCallback
+# @param {(Error|null)} error - error
+# @param {EnvironmentVariable[]} environmentVariables - environment variables
+###
+
+###*
+# @summary Get all device environment variables
+# @public
+# @function
+#
+# @param {(String|Number)} deviceId - device id
+# @param {module:resin.models.environment-variables.device~getAllCallback} callback - callback
+#
+# @example
+#	resin.models.environmentVariables.device.getAll 51, (error, environmentVariables) ->
+#		throw error if error?
+#		console.log(environmentVariables)
+###
+exports.device.getAll = (deviceId, callback) ->
+	return pine.get
+		resource: 'device_environment_variable'
+		options:
+			filter:
+				device: deviceId
+			expand: 'device'
+			orderby: 'env_var_name asc'
+	.tap (environmentVariables) ->
+		if _.isEmpty(environmentVariables)
+			throw new errors.ResinNotAny('device environment variables')
+	.nodeify(callback)
+
+###*
+# device.create callback
+# @callback module:resin.models.environment-variables.device~createCallback
+# @param {(Error|null)} error - error
+###
+
+###*
+# @summary Create a device environment variable
+# @public
+# @function
+#
+# @param {(String|Number)} deviceId - device id
+# @param {String} name - environment variable name
+# @param {String} value - environment variable value
+# @param {module:resin.models.environment-variables.device~createCallback} callback - callback
+#
+# @example
+#	resin.models.environmentVariables.device.create 51, 'EDITOR', 'vim', (error) ->
+#		throw error if error?
+###
+exports.device.create = (deviceId, name, value, callback) ->
+	return pine.post
```

Comment:
@emirotin :) Thanks for your help. I implement support for "data" as well in resin-pine, to avoid a similar issue in the future: https://github.com/resin-io/resin-pine/pull/3.